### PR TITLE
ci(changelog): add a tool to auto update changelog from releases

### PR DESCRIPTION
**Pull request by lahabana**
- We always read whatever is in the release and generate the changelog.md
thanks to this
- Add an action that opens a PR to update the changelog whenever a release changes

Signed-off-by: Charly Molter <charly.molter@konghq.com>

### Checklist prior to review

- [x] Link to docs PR or issue -- N/A
- [x] Link to UI issue or PR -- N/A
- [x] Is the [issue worked on linked][1]? -- #4472
- [x] Unit Tests -- NA
- [x] E2E Tests -- NA
- [x] Manual Universal Tests -- NA
- [x] Manual Kubernetes Tests -- NA
- [x] Do you need to update [`UPGRADE.md`](/UPGRADE.md)? -- NA
- [x] Does it need to be backported according to the [backporting policy](/CONTRIBUTING.md#backporting)? -- NA

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: .github/workflows/update-changelog.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+jobs:
+  upgrade-kuma:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: kong-mesh
+    steps:
+      - name: "Clone Kuma"
+        uses: actions/checkout@v2
+
```
Comment (michaelbeaumont) [inline .github/workflows/update-changelog.yaml:16]: Maybe use the github app? I just added the secrets
https://github.com/kumahq/kuma-website/blob/master/.github/workflows/sync-generated.yml#L10-L15
https://github.com/kumahq/kuma-website/blob/master/.github/workflows/sync-generated.yml#L33

Comment (lahabana) [reply .github/workflows/update-changelog.yaml:16]: makes snese!

File: .github/workflows/update-changelog.yaml [hunk 2]
```diff
@@ -23,10 +28,11 @@ jobs:
...
-          committer: GitHub <noreply@github.com>
-          author: github-actions[bot] <github-actions[bot]@users.noreply.github.com>
           signoff: true
           branch: chore/update-changelog
           delete-branch: true
           title: "docs(CHANGELOG.md): Updating changelog"
           draft: false
+          token: ${{ steps.github-app-token.outputs.token }}
+          committer: kumahq[bot] <110050114+kumahq[bot]@users.noreply.github.com>
+          author: kumahq[bot] <110050114+kumahq[bot]@users.noreply.github.com>
```
Comment (michaelbeaumont) [inline .github/workflows/update-changelog.yaml:38]: One annoying thing about this is that for some reason it adds the `coauthor-by` tag to the commit for some reason, even though committer and author are the same... Not sure how to get rid of that.

**Review by michaelbeaumont — APPROVED** [review]
Cool!

**Comment by lahabana** [comment]
Next step is going to create the draft release with a pregen changelog so we can smoothen this story

**Pull request merged**